src/backend/api/Fusion.Resources.Api/FusionEvents/LineOrgSyncronizationHandler.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Fusion.Events;
+using System.Threading;
+using System.Threading.Tasks;
+using Fusion.Resources.Domain;
+using Newtonsoft.Json;
+using MediatR;
+using System;
+using System.Linq;
+
+namespace Fusion.Resources.Api
+{
+    /// <summary>
+    /// Handler to execute syncronisation events from the line org service. 
+    /// We only want updates to be executed by a single instance in case of multiple instances running the api. 
+    /// 
+    /// The handler should accept events, collect relevant data, determine if event is relevant and dispatch events the domain should handle to perform updates.
+    /// 
+    /// Should consider what the handler actually does (running in context of the event execution) or if some work could be dispatched to an async handler (e.g. service bus). 
+    /// If the handler throws an unhandled exception, the whole processing will be terminated and re-executed as the message is retried. If it consistently fail it will fall to a dead letter queue.
+    /// 
+    /// </summary>
+    public class LineOrgSyncronizationHandler : ISubscriptionHandler
+    {
+        private readonly IMediator mediator;
+
+        public LineOrgSyncronizationHandler(IMediator mediator)
+        {
+            this.mediator = mediator;
+        }
+
+        public async Task ProcessMessageAsync(MessageContext ctx, string? body, CancellationToken cancellationToken)
+        {
+
+            // Process the even 
+            switch (ctx.Event.Type)
+            {
+                case var org when org == LineOrgEventTypes.OrgUnit.Name:
+                    await HandleOrgUnitEventAsync(body);
+                    break;
+            }
+        }
+
+        private async Task HandleOrgUnitEventAsync(string? body)
+        {
+            var payloadData = JsonConvert.DeserializeObject<LineOrgEventBody>(body ?? "");
+
+            if (payloadData is null)
+            {
+                return;
+            }
+
+            if (payloadData.GetChangeType() == LineOrgEventBody.ChangeType.Updated)
+            {
+                if (payloadData.Changes?.Any(i => i.EqualsIgnCase("fullDepartment")) == true) 
+                {
+                    await HandleOrgUnitUpdatedAsync(payloadData);
+                }
+            }
+        }
+
+        private async Task HandleOrgUnitUpdatedAsync(LineOrgEventBody payloadData)
+        {            
+            // Need to disable cache 
+            using var disableCacheScope = new Fusion.Integration.DisableCacheScope();   // Need to verify this works..
```

Comment:
regarding the comment here, did this work? Or do we need to check this in CI?

Reply:
Will fix comment in next pr. Have confirmed that the cache invalidation is used in the integration lib org resolver.